subxt/src/rpc/types.rs

```diff
@@ -239,30 +239,41 @@ pub type SystemProperties = serde_json::Map<String, serde_json::Value>;
 ///
 /// This is copied from `sp-transaction-pool` to avoid a dependency on that crate. Therefore it
 /// must be kept compatible with that type from the target substrate version.
+///
+/// Substrate produces `camelCase` events, while smoldot produces `CamelCase` events.
```

Comment:
Smoldot is not supposed to produce `CamelCase` events

Reply:
Opened https://github.com/smol-dot/smoldot/issues/610

Reply:
I didn't realize when I read this comment, but it's smoldot that generates `transaction_unstable_watchEvent` events when it shouldn't.

I've noticed this independently, and this has now been fixed in https://github.com/smol-dot/smoldot/pull/677

Reply:
Thanks a lot for fixing this! It will make the code simpler from our side!

Reply:
This is not really my problem, but my two is I'd suggest you update smoldot and revert these changes. There's no need to leave code that bypasses smoldot bugs when these bugs have been fixed.